compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -178,6 +205,7 @@ generateSrcFromLf thisPkgId pkgMap m = noLoc mod
     decls =
         concat $ do
             LF.DefDataType {..} <- NM.toList $ LF.moduleDataTypes m
+            guard $ not $ isTypeClass dataCons
```

Comment:
Let's only deal with serializable types. They can be spotted using https://github.com/digital-asset/daml/blob/98a8135cb58b76ad7203cba723c3484077099efe/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs#L613

Reply:
changed.